### ISSUE TITLE
Wrong button placement if similar gates are present

### ISSUE THREAD

**Issue by yisharsr** [stargate-rewritten/Stargate-Bukkit#296]
### Bug Description

The button is placed in wrong positions when multiple gate configurations exist with the same materials

### Reproduction Steps

I have two similar but different sizes custom gate files:

```
portal-open=END_GATEWAY
portal-closed=AIR
button=STONE_BUTTON
toowner=false
X=AMETHYST_BLOCK
-=AMETHYST_BLOCK

 XX 
X..X
-..-
X*.X
 XX 
```

```
portal-open=END_GATEWAY
portal-closed=AIR
button=STONE_BUTTON
toowner=false
X=AMETHYST_BLOCK
-=AMETHYST_BLOCK

 X 
X.X
-.-
X*X
 X 
```

```
portal-open=END_GATEWAY
portal-closed=AIR
button=STONE_BUTTON
toowner=false
X=AMETHYST_BLOCK
-=AMETHYST_BLOCK

 X 
-.-
X*X
 X 
```

 
 As you can see, they are the same but in different sizes, 2x3, 1x2 and 1x3.
 
 But the buttons as you can see in the screenshots end up missplaced in two of the three configs:
 
![image](https://github.com/stargate-rewritten/Stargate-Bukkit/assets/1948585/9cc21fb3-fc40-44e2-ad74-ceab8224c9d5)
![image](https://github.com/stargate-rewritten/Stargate-Bukkit/assets/1948585/705cef9f-4be8-478f-bb68-41e309b48e7f)

Let me know if i´m doing anything wrong but tried a few times and it looks to me more like a bug than a config error.

**Comment by yisharsr** [comment]
I forgot to add, the buttons work as intended, the stargates work, its just the placement that is odd.

**Comment by Thorinwasher** [comment]
Thanks for posting this issue!

An initial question: what version of stargate are you using? If you're using alpha, could you do `sg trace` and post the text in that outputfile in here. Any sensitive information is removed, that is, all mysql stuff

**Comment by yisharsr** [comment]
> Thanks for posting this issue!
> 
> An initial question: what version of stargate are you using? If you're using alpha, could you do `sg trace` and post the text in that outputfile in here. Any sensitive information is removed, that is, all mysql stuff

I´m using version 0.11.5.3 on spigot 1.19.3. sg trace command doesn´t exist (responds /sg <reload/config/about> ...). Can´t find an alpha release and I dont have the tools to build java, but i can try if you point me to a .jar that I can load.

**Comment by Pheotis** [comment]
> > Thanks for posting this issue!
> > An initial question: what version of stargate are you using? If you're using alpha, could you do `sg trace` and post the text in that outputfile in here. Any sensitive information is removed, that is, all mysql stuff
> 
> I´m using version 0.11.5.3 on spigot 1.19.3. sg trace command doesn´t exist (responds /sg <reload/config/about> ...). Can´t find an alpha release and I dont have the tools to build java, but i can try if you point me to a .jar that I can load.

Thanks for that!
We've pushed a fix to legacy (9d48374) that should address this issue.
It will be available in version `0.11.5.4`, which will release later today.

As for the alpha version, although it doesn't have this *particular* bug, it has plenty of others and is generally unstable;
We are not yet confident in its suitability for production, and as such, we have not yet made a public release channel for it.
In the coming months, we will begin to publish rewritten builds once they have reached the beta cycle.

For now, alpha builds are distributed through an alpha testing program via [our discord](https://sgrewritten.org/discord);
If you are interested, you may join by reacting to [this message](https://canary.discord.com/channels/853659042799026177/853659045671337996/978737916782211162) and searching for the `#alpha-testing` channel (builds are pinned)
